src/implementations/osmosis.rs

```diff
@@ -134,6 +124,7 @@ impl Burn for OsmosisDenom {
                     amount: amount.to_string(),
                 }),
                 sender: env.contract.address.to_string(),
+                burn_from_address: env.contract.address.to_string(),
```

Comment:
thsi should work now but a better api long term would be to take in the burn from address. then we don't need to check the funds in the vault. Given that the creator of a denom can burn from another address

Reply:
Yeah I agree but the whole contract needs a refactor so I thought I'd make as few changes as possible for now.